Source/JavaScriptCore/runtime/ISO8601.cpp

```diff
@@ -779,11 +807,36 @@ static std::optional<CalendarRecord> parseCalendar(StringParsingBuffer<Character
 
     if (!canBeCalendar(buffer))
         return std::nullopt;
-    buffer.advanceBy(6);
+    bool isCritical = buffer[1] == '!';
+    // Skip '[' or '[!'
+    buffer.advanceBy(isCritical ? 2 : 1);
+
+    // Parse the key and check if it's equal to "u-ca"
+    unsigned keyLength = 0;
+    while (buffer[keyLength] != '=')
+        keyLength++;
+    Vector<LChar, maxCalendarLength> key(buffer.consume(keyLength));
```

Comment:
Why can't we just have span from buffer?

Reply:
Fixed in [bf03a26](https://github.com/WebKit/WebKit/pull/36361/commits/bf03a263a990d23ef9c258b1248ac67b575d5a70)